buildroot/share/PlatformIO/scripts/common-features-dependencies.py

```diff
@@ -36,6 +36,36 @@ def load_config():
 			else:
 				FEATURE_DEPENDENCIES[key[0].upper()]['lib_deps'] += [dep]
 
+def get_all_known_libs():
+	known_libs = []
+	for feature in FEATURE_DEPENDENCIES:
+		if not 'lib_deps' in FEATURE_DEPENDENCIES[feature]:
+			continue
+		for dep in FEATURE_DEPENDENCIES[feature]['lib_deps']:
+			name, _, _ = PackageManager.parse_pkg_uri(dep)
+			known_libs.append(name)
+	return known_libs
+
+def get_all_env_libs():
+	env_libs = []
+	lib_deps = env.GetProjectOption("lib_deps")
+	for dep in lib_deps:
+		name, _, _ = PackageManager.parse_pkg_uri(dep)
+		env_libs.append(name)
+	return env_libs
+
+# We need to ignore all non-used libs,
+# so if a lib folder lay forgotten in .pio/lib_deps, it
+# will not break compiling
```

Comment:
I'm uncertain why this would ever happen, except as a sub-dependency of another library. The compiler should only add files to the linker stage that were included in the compile stage.

Reply:
PIO still tries to compile non-used libs if they have been downloaded. This is the issue I was seeing in https://github.com/MarlinFirmware/Marlin/pull/18699#issuecomment-660772254 and I had to manually delete the hidden `.pio` folder to make PIO happy/only compile the required libs.

Reply:
But… It should only re-compile them when the `platformio.ini` file changes, because this file can affect compile flags for everything. Otherwise it should be trusting the existing compiled `.o` files.

Reply:
Exactly. But `platformio.ini` will change at least once, when the user pull our last changes. As every lib were downloaded before the build optimization and we removed the lib_ignores, a lot of the users have don't used libs in `.pio/lib_deps` right now.
More than one user already came with the this problem, right after they pull the last changes. After they deleted `.pio/lib_deps` folder, the problem was gone. 

But, it may happen when you are just testing features. The libs will be downloaded and it will affect your build, and even give a error, if not be ignored... :-/

So, this PR is just a kind of reinforcement to not break builds, mainly for the ones are just pulling the lasts changes.